internal/initwd/module_install.go

```diff
@@ -49,6 +49,28 @@ type moduleVersion struct {
 	version string
 }
 
+// TypeModuleVersionDeprecation is a constant of TypeDiagnosticExtra indicating
+// that a Warning is related to a deprecated module version is in use. This constant is part of the public API and cannot be edited/removed without correcsponding changes in TFC agent.
+type TypeDiagnosticExtra string
+
+const TypeModuleVersionDeprecation = "module_version_deprecation"
+
+// ModuleVersionDeprecationDiagnosticExtra holds the diagnostic information
+// about the deprecation of a module version for later use in the building the warnings in TFC.
+// This struct is part of the public API and cannot be edited/removed without correcsponding changes in TFC agent.
```

Comment:
Don't mention TFC or TFC agent -- also "correcsponding"

```suggestion
// about the deprecation of a module version. This ends up being serialized as extra data within a
// diagnostic and should be considered public API.
```

Reply:
Made an edit to the commit https://github.com/hashicorp/terraform/pull/35231/commits/1368c93862d046729782fc91107ce683191be0c3